src/main/java/com/google/sps/data/DatabaseHandler.java

```diff
@@ -42,6 +50,31 @@ public void addBookStockToDatastore(String bookId) {
       datastore.put(libraryEntity);
     }
   }
+
+  /** 
+   * Return the library Entity with id libraryId from datastore, or null if there is no library 
+   * with id libraryId in datastore.
+   */
+  private Entity getLibraryEntity(int libraryId) {
+    return datastore.prepare(new Query("Library")
+          .setFilter(new Query.FilterPredicate("libraryId", Query.FilterOperator.EQUAL, libraryId)))
+          .asSingleEntity();
+  }
+
+  /**
+   * Adds the library with id libraryId only if it wasn't already added to datastore. 
+   */
+  private void addLibraryToDatastore(int libraryId, LatLng libraryLoc) {
+    if (getLibraryEntity(libraryId) != null) {
+      // The library has already been added to datastore.
+      return ;
+    }
+    Entity libraryEntity = new Entity("Library");
```

Comment:
Optional nit: the string `"Library"` is used in multiple places here - might make sense to convert it to a constant.

Reply:
Done.